support/tests/test_tealium_lifecycle/TealiumLifecycleIntegrationTests.swift

```diff
@@ -0,0 +1,134 @@
+        let end = Date(timeIntervalSince1970: 1480557600)       // 2016 DEC 2 - 02:00 UTC
+        let sessionSuccess = TealiumLifecycleSession(withWakeDate: start)
+        sessionSuccess.sleepDate = end
+        let sessionCrashed = TealiumLifecycleSession(withWakeDate: start)
+        
+        let lifecycle = TealiumLifecycle()
+        let _ = lifecycle.newLaunch(atDate: start, overrideSession: nil)
+
+        // Double checking that we aren't returning "true" if we're still in the first launch session.
+        let initialDetection = lifecycle.newCrashDetected()
+        XCTAssert(initialDetection == nil, "")
+        
+        // Check if first launch session resulted in a crash on subsequent launch
+        let _ = lifecycle.newLaunch(atDate: start, overrideSession: nil)
+        XCTAssert(lifecycle.newCrashDetected() == "true", "Should have logged crash as initial launch did not have sleep data. FirstSession: \(lifecycle.sessions.first)")
+        
+        lifecycle.sessions.first?.sleepDate = end
+        XCTAssert(lifecycle.newCrashDetected() == nil, "Should not have logged crash as initial launch has sleep data. SessionFirst: \(lifecycle.sessions.first) \nall sessions:\(lifecycle.sessions)")
+        
+        
+        lifecycle.sessions.append(sessionCrashed)
+        let _ = lifecycle.newLaunch(atDate: Date(), overrideSession: nil)
+        XCTAssertTrue(lifecycle.newCrashDetected() == "true", "Crashed prior session not caught. Sessions: \(lifecycle.sessions)")
+        
+    }
+
+ 
+    func dictionaryFromJSONFile(withName: String) -> [String:Any]? {
+        
+        let bundle = Bundle(for: type(of:self))
+        
+        guard let path = bundle.path(forResource: withName, ofType: "json") else {
+            assertionFailure("Target json file with name:\(withName) not found.")
+            return nil
+        }
+        
+        do {
+            let jsonData = try NSData(contentsOfFile: path, options: NSData.ReadingOptions.mappedIfSafe)
+            do {
+                let jsonResult: NSDictionary = try JSONSerialization.jsonObject(with: jsonData as Data, options: JSONSerialization.ReadingOptions.mutableContainers) as! NSDictionary
+                guard let jsonDictionary = jsonResult as? [String:Any] else {
+                    assertionFailure("Target json file with name:\(withName) could not be converted to [String:Any].")
+                    return nil
+                }
+                return jsonDictionary
+            } catch {
+                // Could not convert JSON file to dictionary
+                assertionFailure("Target json file with name:\(withName) could not be converted to an NSDictionary.")
+                return nil
+            }
+        } catch {
+            // Could not open file at path as NSData
+            assertionFailure("Target json file with name:\(withName) could not be opened as NSData.")
+            return nil
+        }
```

Comment:
This was originally in the test_helper.  Does not work correctly there due to the bundle identifier api.